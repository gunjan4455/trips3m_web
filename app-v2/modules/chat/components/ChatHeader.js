import React from 'react';
import PropTypes from 'prop-types';

import AgentOnlineIndicator from 'modules/shared/agentProfile/AgentOnlineIndicator';
import { formatChatKeys } from 'helpers/rtms';
import RequestCallback from 'modules/requestCallback/index';

import styled from '@emotion/styled';

const GChatHeadDiv = styled.div({
  maxWidth: 'calc(100% - 42px)',
});

class ChatHeader extends React.Component {
  static propTypes = {
    chatKeys: PropTypes.object.isRequired,
    agentId: PropTypes.number.isRequired,
    agentName: PropTypes.string.isRequired,
    userDetails: PropTypes.object.isRequired,
  };

  render() {
    const { agentId, agentName, chatKeys, userDetails } = this.props;

    return (
      <div className={`row row- chat-header relative at_chatheader flex`}>
        <GChatHeadDiv className="flexFull p0 mr8">
          <p className="m0 pb4 sfcw f16 ellipsis">{agentName}</p>
          <div className="clearfix relative">
            <AgentOnlineIndicator
              showPresenceMessage={true}
              userId={agentId}
              pubnubConfig={formatChatKeys(chatKeys)}
              userDetails={userDetails}
              isChatIndicator={true}
            />
          </div>
        </GChatHeadDiv>
        <RequestCallback isIcononly={true} isWhiteIcon={true} />
      </div>
    );
  }
}

export default ChatHeader;
