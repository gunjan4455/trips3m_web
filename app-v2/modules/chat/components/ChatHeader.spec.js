import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import ChatHeader from './ChatHeader';

configure({ adapter: new Adapter() });

const sendMessageMock = jest.fn();

const DEFAULT_PROPS = {
  chatKeys: {
    loaded: true,
    channel: 'Chat',
    publishKey: '12121',
    subscribeKey: '121212'
  },
  agentId: 1111111,
  agentName: 'Test',
  userDetails: {
    userId: 111111
  }
};

const shallowWrappperChatHeader = (props) => {
  return shallow(<ChatHeader {...props} />);
};

describe('ChatHeader: Render Correctly ', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should match with the previous snapshot', () => {
    const wrapper = shallowWrappperChatHeader(props);
    expect(wrapper).toMatchSnapshot();
  });

});
