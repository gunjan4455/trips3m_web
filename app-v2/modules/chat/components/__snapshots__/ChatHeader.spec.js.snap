// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHeader: Render Correctly  should match with the previous snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChatHeader
    agentId={1111111}
    agentName="Test"
    chatKeys={
      Object {
        "channel": "Chat",
        "loaded": true,
        "publishKey": "12121",
        "subscribeKey": "121212",
      }
    }
    userDetails={
      Object {
        "userId": 111111,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(render)
          className="flexFull p0 mr8"
        >
          <p
            className="m0 pb4 sfcw f16 ellipsis"
          >
            Test
          </p>
          <div
            className="clearfix relative"
          >
            <AgentOnlineIndicator
              isChatIndicator={true}
              pubnubConfig={
                Object {
                  "channels": Object {
                    "chat": "Chat",
                  },
                  "publish_key": "12121",
                  "subscribe_key": "121212",
                }
              }
              showPresenceMessage={true}
              userDetails={
                Object {
                  "userId": 111111,
                }
              }
              userId={1111111}
            />
          </div>
        </ForwardRef(render)>,
        <withRouter(Connect(RequestCallback))
          isIcononly={true}
          isWhiteIcon={true}
        />,
      ],
      "className": "row row- chat-header relative at_chatheader flex",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <p
              className="m0 pb4 sfcw f16 ellipsis"
            >
              Test
            </p>,
            <div
              className="clearfix relative"
            >
              <AgentOnlineIndicator
                isChatIndicator={true}
                pubnubConfig={
                  Object {
                    "channels": Object {
                      "chat": "Chat",
                    },
                    "publish_key": "12121",
                    "subscribe_key": "121212",
                  }
                }
                showPresenceMessage={true}
                userDetails={
                  Object {
                    "userId": 111111,
                  }
                }
                userId={1111111}
              />
            </div>,
          ],
          "className": "flexFull p0 mr8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
              "className": "m0 pb4 sfcw f16 ellipsis",
            },
            "ref": null,
            "rendered": "Test",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <AgentOnlineIndicator
                isChatIndicator={true}
                pubnubConfig={
                  Object {
                    "channels": Object {
                      "chat": "Chat",
                    },
                    "publish_key": "12121",
                    "subscribe_key": "121212",
                  }
                }
                showPresenceMessage={true}
                userDetails={
                  Object {
                    "userId": 111111,
                  }
                }
                userId={1111111}
              />,
              "className": "clearfix relative",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "isChatIndicator": true,
                "pubnubConfig": Object {
                  "channels": Object {
                    "chat": "Chat",
                  },
                  "publish_key": "12121",
                  "subscribe_key": "121212",
                },
                "showPresenceMessage": true,
                "userDetails": Object {
                  "userId": 111111,
                },
                "userId": 1111111,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__emotion_base": "div",
          "__emotion_forwardProp": undefined,
          "__emotion_real": [Circular],
          "__emotion_styles": Array [
            Object {
              "maxWidth": "calc(100% - 42px)",
            },
          ],
          "defaultProps": undefined,
          "displayName": "Styled(div)",
          "render": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "isIcononly": true,
          "isWhiteIcon": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(render)
            className="flexFull p0 mr8"
          >
            <p
              className="m0 pb4 sfcw f16 ellipsis"
            >
              Test
            </p>
            <div
              className="clearfix relative"
            >
              <AgentOnlineIndicator
                isChatIndicator={true}
                pubnubConfig={
                  Object {
                    "channels": Object {
                      "chat": "Chat",
                    },
                    "publish_key": "12121",
                    "subscribe_key": "121212",
                  }
                }
                showPresenceMessage={true}
                userDetails={
                  Object {
                    "userId": 111111,
                  }
                }
                userId={1111111}
              />
            </div>
          </ForwardRef(render)>,
          <withRouter(Connect(RequestCallback))
            isIcononly={true}
            isWhiteIcon={true}
          />,
        ],
        "className": "row row- chat-header relative at_chatheader flex",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p
                className="m0 pb4 sfcw f16 ellipsis"
              >
                Test
              </p>,
              <div
                className="clearfix relative"
              >
                <AgentOnlineIndicator
                  isChatIndicator={true}
                  pubnubConfig={
                    Object {
                      "channels": Object {
                        "chat": "Chat",
                      },
                      "publish_key": "12121",
                      "subscribe_key": "121212",
                    }
                  }
                  showPresenceMessage={true}
                  userDetails={
                    Object {
                      "userId": 111111,
                    }
                  }
                  userId={1111111}
                />
              </div>,
            ],
            "className": "flexFull p0 mr8",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Test",
                "className": "m0 pb4 sfcw f16 ellipsis",
              },
              "ref": null,
              "rendered": "Test",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <AgentOnlineIndicator
                  isChatIndicator={true}
                  pubnubConfig={
                    Object {
                      "channels": Object {
                        "chat": "Chat",
                      },
                      "publish_key": "12121",
                      "subscribe_key": "121212",
                    }
                  }
                  showPresenceMessage={true}
                  userDetails={
                    Object {
                      "userId": 111111,
                    }
                  }
                  userId={1111111}
                />,
                "className": "clearfix relative",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "isChatIndicator": true,
                  "pubnubConfig": Object {
                    "channels": Object {
                      "chat": "Chat",
                    },
                    "publish_key": "12121",
                    "subscribe_key": "121212",
                  },
                  "showPresenceMessage": true,
                  "userDetails": Object {
                    "userId": 111111,
                  },
                  "userId": 1111111,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__emotion_base": "div",
            "__emotion_forwardProp": undefined,
            "__emotion_real": [Circular],
            "__emotion_styles": Array [
              Object {
                "maxWidth": "calc(100% - 42px)",
              },
            ],
            "defaultProps": undefined,
            "displayName": "Styled(div)",
            "render": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isIcononly": true,
            "isWhiteIcon": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
